### PR TITLE
call `super().__post_init__` in `ForeachFuncinfo.__post_init__`

### LINKED ISSUE #125295: `OpInfo.supported_dtypes` could return `None`
this would occur when `dtypesIfCUDA` or `dtypesIfROCM` are not set: https://github.com/pytorch/pytorch/blob/c3c4465f504282f61f962027adf2bb9ea196b976/torch/testing/_internal/opinfo/core.py#L1342-L1348

while https://github.com/pytorch/pytorch/blob/c3c4465f504282f61f962027adf2bb9ea196b976/torch/testing/_internal/opinfo/core.py#L717-L727 seems to me that `OpInfo`'s post init or dunder init would use and set  `dtypes` as `dtypesIfCUDA` and/or `dtypesIfROCM`.

ref: https://github.com/pytorch/pytorch/actions/runs/8900253100/job/24444527860#step:15:1514 of https://hud.pytorch.org/pr/pytorch/pytorch/125001#24444527860 where `dtypesIfROCM` is not set.

### PR DESCRIPTION
**Pull request by crcrpar**
obviously the current main branch's `ForeachFuncInfo`'s dunder post init doesn't `super().__post_init__()` which does some setup including setting `dtypesIfCUDA` and `dtypesIfROCM`.

Fixes #125295
related: #125001 

cc @jeffdaily

### CODE REVIEW

File: torch/testing/_internal/opinfo/core.py [hunk 1]
```diff
@@ -2750,6 +2750,9 @@ def __post_init__(self):
             self.ref = torch.clamp_min
             self.ref_inplace = torch.Tensor.clamp_min_
 
+        # The following sets `dtypesIfCUDA` and `dtypesIfROCM` accordingly.
+        super().__post_init__()
```
Comment (crcrpar) [inline torch/testing/_internal/opinfo/core.py:2754]: my bad, apologies

**Review by janeyx99 — APPROVED** [review]
ship it!

**Pull request closed without merging**